Allphi/Domain/Domaincontroller.cs

```diff
@@ -73,17 +74,127 @@ public ParkingSpot GetAvailableParkingSpotVisitor()
+            ParkingSpot parkingSpot = _parkingRepo.GetAvailableParkingSpotReserved(employee.Business);
+            if (parkingSpot == null) parkingSpot = _parkingRepo.GetAvailableParkingSpotUnreserved();
+            if (IsLicensePlateValid(licensePlate) && parkingSpot != null &&
+                _parkingRepo.CountParkingSpotByPlate(license) == 0)
+            {
+                parkingSpot.Plate = license;
+                _parkingRepo.UpdateParkingSpot(parkingSpot);
+            }
+        }
+
+        #endregion Parking
+
+        #region Business
+
+        public void CreateBusiness(string name, string? address, string? phone, string email, string btw)
+        {
+            _businessRepo.CreateBusiness(new Business(name, btw, email, address, phone));
+        }
+
+        public List<Business> GetBusinesses()
+        {
+            return _businessRepo.GetBusinesses();
+        }
+
+        #endregion Business
+
+        public List<Visitor> GetVisitors()
+        {
+            return _visitorRepo.GetVisitors();
+        }
+
+        public List<Contract> GetContracts()
+        {
+            return _contractRepo.GetContracts();
+        }
+
+        public List<Employee> GetEmployees()
+
+        {
+            return _employeeRepo.GetEmployees();
+        }
+
+        public List<string> ConvertBusinessToStringList(object input)
+        {
+            Business selected = (Business)input;
+            List<string> business = new List<string>();
+            business.Add(selected.Name);
+            business.Add(selected.Address);
+            business.Add(selected.Phone);
+            business.Add(selected.Email);
+            business.Add(selected.Btw);
+            return business;
+        }
+
+        public List<string> ConvertVisitorToStringList(object input)
+        {
+            Visitor selected = (Visitor)input;
+            List<string> visitor = new List<string>();
+            visitor.Add(selected.Name);
+            visitor.Add(selected.Email);
+            visitor.Add(selected.Plate);
+            visitor.Add(selected.Business);
+            return visitor;
+        }
+
+        public void CreateVisitor(string name, string email, string? plate, string business)
+        {
+            _visitorRepo.CreateVisitor(new Visitor(name, email, business, plate));
+        }
+
+        public List<string> ConvertContractToStringList(object selectedItem)
+        {
+            Contract selected = (Contract)selectedItem;
+            List<string> contract = new List<string>();
+            contract.Add(selected.Business.Name);
+            contract.Add(selected.TotalSpaces.ToString());
+            contract.Add(selected.StartDate.ToString());
+            contract.Add(selected.EndDate.ToString());
+            return contract;
+        }
+
+        public void CreateContract(string spots, string business, DateTime start, DateTime end)
+        {
+            int totalSpots = int.Parse(spots);
```

Comment:
Logic related to the UI layer only should be in the UI layer.
Also this will crash given a non parseable int.